changes.go

```diff
@@ -351,8 +351,10 @@ type AddApplicationChange struct {
 	Params AddApplicationParams
 }
 
-// GUIArgs implements Change.GUIArgs.
-func (ch *AddApplicationChange) GUIArgs() []interface{} {
+// GUIArgsWithDevices implements Change.GUIArgs for backward support with minimun effort.
+// Because bundlechanges is shared across juju repositories and there is no proper versioning here, so implement GUIArgs
```

Comment:
I would have just said that this does the same as GUIArgs and also adds devies support. Everything else is irrelevant as a comment on the method :)